### PR TITLE
feat: implement availability per endpoint and availability per gateway widgets

### PR DESCRIPTION
**Pull request by JedrzejJanasiak**
## Issue

https://gravitee.atlassian.net/browse/APIM-7428
https://gravitee.atlassian.net/browse/APIM-7429

## Description

Implementation of two health check widgets for API
- Availability per endpoint
- Availability per gateway

## Additional context
Zrzut ekranu 2024-11-13 o 17 44 08

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-tybplleyaa.chromatic.com)

### CODE REVIEW

File: gravitee-apim-console-webui/src/entities/management-api-v2/api/v4/healthCheck.fixture.ts [hunk 1]
```diff
@@ -32,30 +32,21 @@ export function fakeApiHealthResponseTimeOvertime(attribute?: Partial<ApiHealthR
   };
 }
 
-export function fakeApiHealthAvailability(attribute?: Partial<ApiAvailability>): ApiAvailability {
```
Comment (pes-spyro-soft-com) [inline gravitee-apim-console-webui/src/entities/management-api-v2/api/v4/healthCheck.fixture.ts:35]: we keep this functions with attribute paramiters so we can manipulate result of test data. Its a pattern here.

File: gravitee-apim-console-webui/src/entities/management-api-v2/api/v4/healthCheck.fixture.ts [hunk 2]
```diff
@@ -32,30 +32,21 @@ export function fakeApiHealthResponseTimeOvertime(attribute?: Partial<ApiHealthR
...
-  const base: ApiAvailability = {
-    global: 0.9876,
-    group: {
-      example: 100,
```
Comment (pes-spyro-soft-com) [inline gravitee-apim-console-webui/src/entities/management-api-v2/api/v4/healthCheck.fixture.ts:39]: you can just replace "example field" with your someSampleGroup, no need to modify function as a whole.

File: gravitee-apim-console-webui/src/management/api/health-check-dashboard-v4/components/availability-per-field/availability-per-field.component.ts [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    GioTableWrapperModule,
+    MatTable,
+    MatSort,
+    MatColumnDef,
+    MatCell,
+    MatCellDef,
+    MatHeaderCell,
+    MatHeaderCellDef,
+    MatHeaderRow,
+    MatHeaderRowDef,
```
Comment (pes-spyro-soft-com) [inline gravitee-apim-console-webui/src/management/api/health-check-dashboard-v4/components/availability-per-field/availability-per-field.component.ts]: some of the imports can be reduced by importing module containing them.

**Review by pes-spyro-soft-com — DISMISSED** [review]

**Review by pes-spyro-soft-com — COMMENTED** [review]
need some work

File: gravitee-apim-console-webui/src/entities/management-api-v2/api/v4/healthCheck.fixture.ts (hunk 1 above)
Comment (JedrzejJanasiak) [reply gravitee-apim-console-webui/src/entities/management-api-v2/api/v4/healthCheck.fixture.ts:35]: done 👍  however, I still don't get why we use this pattern here if we don't make use of it 🤔

Comment (pes-spyro-soft-com) [reply gravitee-apim-console-webui/src/entities/management-api-v2/api/v4/healthCheck.fixture.ts:35]: cool, I will use it :-)

File: gravitee-apim-console-webui/src/entities/management-api-v2/api/v4/healthCheck.fixture.ts (hunk 2 above)
Comment (JedrzejJanasiak) [reply gravitee-apim-console-webui/src/entities/management-api-v2/api/v4/healthCheck.fixture.ts:39]: I can't 😞 That solution you propose doesn't work in this case. I started with it and had to find another way of creating a dictionary to make this fixture work.

File: gravitee-apim-console-webui/src/management/api/health-check-dashboard-v4/components/availability-per-field/availability-per-field.component.ts (hunk 3 above)
Comment (JedrzejJanasiak) [reply gravitee-apim-console-webui/src/management/api/health-check-dashboard-v4/components/availability-per-field/availability-per-field.component.ts]: done 👍

File: gravitee-apim-console-webui/src/management/api/health-check-dashboard-v4/components/availability-per-field/availability-per-field.component.ts [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    TitleCasePipe,
+  ],
+  selector: 'availability-per-field',
+  standalone: true,
+  styleUrl: './availability-per-field.component.scss',
+  templateUrl: './availability-per-field.component.html',
+})
+export class AvailabilityPerFieldComponent implements OnInit {
+  @Input()
+  field: 'endpoint' | 'gateway';
```
Comment (pes-spyro-soft-com) [inline gravitee-apim-console-webui/src/management/api/health-check-dashboard-v4/components/availability-per-field/availability-per-field.component.ts]: preferable way would be to have enum to use it across feature.

**Review by pes-spyro-soft-com — APPROVED** [review]

File: gravitee-apim-console-webui/src/management/api/health-check-dashboard-v4/components/availability-per-field/availability-per-field.component.ts (hunk 4 above)
Comment (JedrzejJanasiak) [reply gravitee-apim-console-webui/src/management/api/health-check-dashboard-v4/components/availability-per-field/availability-per-field.component.ts]: done

File: gravitee-apim-console-webui/src/services-ngx/api-health-v2.service.ts [hunk 5]
```diff
@@ -49,13 +54,13 @@ export class ApiHealthV2Service {
     return this.httpClient.get<ApiHealthResponseTimeOvertime>(url);
   }
 
-  public getApiAvailability(apiId: string, from: number, to: number): Observable<ApiAvailability> {
-    const url = `${this.constants.env.v2BaseURL}/apis/${apiId}/health/availability?from=${from}&to=${to}&field=endpoint`;
+  public getApiAvailability(apiId: string, from: number, to: number, field: 'endpoint' | 'gateway'): Observable<ApiAvailability> {
```
Comment (pes-spyro-soft-com) [inline gravitee-apim-console-webui/src/services-ngx/api-health-v2.service.ts:57]: FieldParameter enum ;-)

**Pull request merged**